lib/bamboo/adapters/elastic_email_adapter.ex

```diff
@@ -197,10 +197,42 @@ defmodule Bamboo.ElasticEmailAdapter do
 
   defp put_transactional(map), do: Map.put(map, "isTransactional", true)
 
-  defp put_post_back(%{private: %{elastic_custom_vars: %{post_back: post_back}}} = map),
-    do: Map.put(map, "postBack", post_back)
+  defp put_custom_vars(%{private: %{elastic_custom_vars: custom_vars}} = map) do
```

Comment:
Rename to `put_elastic_send_options`, but let’s make it backwards compatible (this code is untested and written in this box, but should work):

```elixir
defp put_elastic_send_options(%{private: %{elastic_custom_vars: options} = private} = email) do
  private =
    private
    |> Map.delete(:elastic_custom_vars)
    |> Map.put(:elastic_send_options, options)
  put_elastic_send_options(%{email | private: private})
end

defp put_elastic_send_options(%{private: %{elastic_send_options: options}} = email) do
  options =
    options
    |> Enum.map(&custom_var/1)
    |> Enum.reject(&is_nil/1)
    |> Enum.reduce(&Map.merge(&2, &1)) # This should be able to be done as |> Enum.into(%{})
   Map.merge(email, options) # The order here is important.
end

defp put_elastic_send_options(email), do: email
```

Your `Enum.reduce/2` with `Map.merge/2` is probably replaceable as `Enum.into/2`, but you should test that. The second one reverses the order of the merge. As written, it is possible that an email would already contain a value that would overwrite why you put into the send options. This guarantees that the send options will overwrite whatever is in the email.

Reply:
`Enum.into` is definitely cleaner.  Just a note that the `elastic_send_options` only processes a subset of the available options.  Specifically it ignores any values that would be filled in by the regular email fields.

Reply:
Absolutely, and we document it that way.